src/python/pants/backend/java/dependency_inference/java_parser.py

```diff
@@ -98,9 +96,7 @@ async def analyze_java_source_dependencies(
 
     proc = Process(
         argv=[
-            coursier.coursier.exe,
-            "java",
-            "--system-jvm",  # TODO(#12293): use a fixed JDK version from a subsystem.
+            f"{jdk_setup.java_home}/bin/java",
```

Comment:
A tricky bit about this in the context of remote execution is that the JDK selection process isn't guaranteed to run on the same machine as the compile.

So rather than having the `jdk_setup` expose `java_home` as a property, it should probably expose it as a command prefix, which we can hope (or know, locally) will hit the cached pre-selected JDK rather than re-fetching it.

Reply:
> So rather than having the jdk_setup expose java_home as a property, it should probably expose it as a command prefix, which we can hope (or know, locally) will hit the cached pre-selected JDK rather than re-fetching it.

That would require using coursier to invoke the JDK every time?

Reply:
Correct. But it appears to take a few milliseconds, because the coursier binary is native.

Also, that time is moot with nailgun: I'll probably get to that after hours this week.

Reply:
Okay implemented calling into coursier's binary each time.